### PR TITLE
DDF-05424: Enum marionette view forced re-render

### LINKED ISSUE #5424: Map settings dropdown does not propagate selected changes to other visualizations
### Description
If multiple map visualizations are open in Intrigue, the user has opened the `Settings` in both visualizations (this is root cause of the issue) and the user changes `Coordinate Format` under one visualization the change is not reflected in the other.

Screen Shot 2019-10-03 at 5 22 51 PM
Screen Shot 2019-10-03 at 5 23 05 PM

### Steps to Reproduce
1. Add two map `visuals` (2D Map and 3D Map)
2. Click `Settings` under each so that the `Coordinate Format` menu comes up
3. Select a `Coordinate Format` from the dropdown that's different from what is currently selected
4. Check the `Coordinate Format` of the other map

#### Expected behavior :
`Coordinate Format`(s) should match between map visuals

#### Actual behavior:
They don't match

#### Reproduces how often:
100%

#### Platform and environment:
macOs, Java 1.8.0_202, Chrome 77.0.3865.90

### Affects versions
2.19.0-SNAPSHOT

### PR DESCRIPTION
**Pull request by leo-sakh**
#### What does this PR do?
Fixes #5424  

#### Who is reviewing it? 
@bennuttle 
@BernardIgiri 
@jrnorth 
@djblue 
@gjvera 
@lavoywj 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@jrnorth

#### How should this be tested?
Please see #5424 

#### What are the relevant tickets?
Fixes: #5424 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx [hunk 1]
```diff
@@ -32,45 +32,53 @@ type Props = {
...
+
+        const propertyModel = new PropertyModel({
           label: this.props.label,
           enum: this.props.options,
-          value: [this.state.value],
+          value: [this.props.value],
           isEditing: true,
         })
+
+        this.state = { propertyModel: propertyModel }
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx]: :pencil2: Could use shorthand here

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx [hunk 2]
```diff
@@ -32,45 +32,53 @@ type Props = {
...
+      }
       onChange() {
         if (this.props.onChange !== undefined) {
-          this.props.onChange(this.propertyModel.getValue()[0])
+          this.props.onChange(this.state.propertyModel.getValue()[0])
         }
       }
       render() {
+        //Forces re-render otherwise the region doesn't get replaced because the view doesn't change
+        const Container = class extends MarionetteRegionContainer {}
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx]: Can you explain how this line forces a re-render? How is this different than using the MarionetteRegionContainer?

Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx]: React will not re-render components if they had't changed and this creates a new instance every time (changes every time). `MarionetteRegionContainer` only picks up changes to the view: https://github.com/codice/ddf/blob/1c555a07bf08654132823fa9f9a69c207aca4a4d/ui/packages/catalog-ui-search/src/main/webapp/react-component/marionette-region-container/marionette-region-container.tsx#L70 Unfortunately `PropertyView` does not result in a re-render on the property value change.

**Comment by bdeining** [comment]
build now

**Comment by leo-sakh** [comment]
build now

**Comment by jrnorth** [comment]
I can hero this.

**Comment by jrnorth** [comment]
Hero successful. The coordinate format setting is now synchronized between the two maps.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx (hunk 2 above)
Comment (andrewkfiedler) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/enum/enum.tsx]: You can accomplish the same thing (forcing a rerender each time) by setting a key on the MarionetteRegionContainer like Math.random().

**Pull request merged**